### ISSUE TITLE
Mutabilities for a gene == 0 & Vector of lambdas == 0

### ISSUE THREAD

**Issue by FerriolCalvet** [bbglab/omega#20]

**Comment by FerriolCalvet** [comment]
Commenting these two lines forces that whenever the `omega estimator mutabilities` are 0, omega cannot compute any dNdS value and does not output the requested gene, sample, impact combination.

https://github.com/bbglab/omega/blob/b044b0a81495695d884a5aacc40afc7a9e65dc33/src/estimator/assemble.py#L130-L131

**Comment by FerriolCalvet** [comment]
I added some logger warnings and these are the differences between before and after commenting the two lines above:

BEFORE:
```
2024-04-10 17:55:47,648 - WARNING | omega.estimator.assemble - No mutations found for ['P19_0044_BDO_01'], ['nonsense'], ['ARID1A'], filling the counts with 0s.
2024-04-10 17:55:47,649 - WARNING | omega.estimator.assemble - No mutations found for ['P19_0044_BDO_01'], ['essential_splice'], ['ARID1A'], filling the counts with 0s.
2024-04-10 17:55:47,653 - WARNING | omega.estimator.assemble - No mutations found for ['P19_0044_BDO_01'], ['nonsense'], ['CDKN1A'], filling the counts with 0s.
2024-04-10 17:55:47,654 - WARNING | omega.estimator.assemble - No mutations found for ['P19_0044_BDO_01'], ['essential_splice'], ['CDKN1A'], filling the counts with 0s.
2024-04-10 17:55:47,656 - WARNING | omega.estimator.assemble - No mutations found for ['P19_0044_BDO_01'], ['nonsense'], ['CREBBP'], filling the counts with 0s.
2024-04-10 17:55:47,657 - WARNING | omega.estimator.assemble - No mutations found for ['P19_0044_BDO_01'], ['essential_splice'], ['CREBBP'], filling the counts with 0s.
2024-04-10 17:55:47,659 - WARNING | omega.estimator.assemble - No mutations found for ['P19_0044_BDO_01'], ['essential_splice'], ['EP300'], filling the counts with 0s.
2024-04-10 17:55:47,661 - WARNING | omega.estimator.assemble - No mutations found for ['P19_0044_BDO_01'], ['missense'], ['FGFR3'], filling the counts with 0s.
2024-04-10 17:55:47,662 - WARNING | omega.estimator.assemble - No mutations found for ['P19_0044_BDO_01'], ['nonsense'], ['FGFR3'], filling the counts with 0s.
2024-04-10 17:55:47,663 - WARNING | omega.estimator.assemble - No mutations found for ['P19_0044_BDO_01'], ['essential_splice'], ['FGFR3'], filling the counts with 0s.
2024-04-10 17:55:47,664 - WARNING | omega.estimator.assemble - No mutations found for ['P19_0044_BDO_01'], ['missense', 'nonsense'], ['FGFR3'], filling the counts with 0s.
2024-04-10 17:55:47,665 - WARNING | omega.estimator.assemble - No mutations found for ['P19_0044_BDO_01'], ['missense', 'nonsense', 'essential_splice'], ['FGFR3'], filling the counts with 0s.
2024-04-10 17:55:47,666 - WARNING | omega.estimator.assemble - No mutations found for ['P19_0044_BDO_01'], ['nonsense'], ['FOXQ1'], filling the counts with 0s.
2024-04-10 17:55:47,667 - WARNING | omega.estimator.assemble - No mutations found for ['P19_0044_BDO_01'], ['essential_splice'], ['FOXQ1'], filling the counts with 0s.
2024-04-10 17:55:47,668 - WARNING | omega.estimator.assemble - Lambda tensors are 0, we are ignoring this case ['P19_0044_BDO_01'], ['essential_splice'], ['FOXQ1'].
2024-04-10 17:55:47,670 - WARNING | omega.estimator.assemble - No mutations found for ['P19_0044_BDO_01'], ['essential_splice'], ['KDM6A'], filling the counts with 0s.
2024-04-10 17:55:47,672 - WARNING | omega.estimator.assemble - No mutations found for ['P19_0044_BDO_01'], ['nonsense'], ['KMT2C'], filling the counts with 0s.
2024-04-10 17:55:47,678 - WARNING | omega.estimator.assemble - No mutations found for ['P19_0044_BDO_01'], ['essential_splice'], ['NOTCH2'], filling the counts with 0s.
2024-04-10 17:55:47,680 - WARNING | omega.estimator.assemble - No mutations found for ['P19_0044_BDO_01'], ['nonsense'], ['PIK3CA'], filling the counts with 0s.
2024-04-10 17:55:47,681 - WARNING | omega.estimator.assemble - No mutations found for ['P19_0044_BDO_01'], ['essential_splice'], ['PIK3CA'], filling the counts with 0s.
2024-04-10 17:55:47,683 - WARNING | omega.estimator.assemble - No mutations found for ['P19_0044_BDO_01'], ['missense'], ['RB1'], filling the counts with 0s.
2024-04-10 17:55:47,683 - WARNING | omega.estimator.assemble - No mutations found for ['P19_0044_BDO_01'], ['nonsense'], ['RB1'], filling the counts with 0s.
2024-04-10 17:55:47,684 - WARNING | omega.estimator.assemble - No mutations found for ['P19_0044_BDO_01'], ['essential_splice'], ['RB1'], filling the counts with 0s.
2024-04-10 17:55:47,685 - WARNING | omega.estimator.assemble - No mutations found for ['P19_0044_BDO_01'], ['missense', 'nonsense'], ['RB1'], filling the counts with 0s.
2024-04-10 17:55:47,686 - WARNING | omega.estimator.assemble - No mutations found for ['P19_0044_BDO_01'], ['missense', 'nonsense', 'essential_splice'], ['RB1'], filling the counts with 0s.
2024-04-10 17:55:47,687 - WARNING | omega.estimator.assemble - No mutations found for ['P19_0044_BDO_01'], ['nonsense'], ['RBM10'], filling the counts with 0s.
2024-04-10 17:55:47,688 - WARNING | omega.estimator.assemble - No mutations found for ['P19_0044_BDO_01'], ['essential_splice'], ['RBM10'], filling the counts with 0s.
2024-04-10 17:55:47,690 - WARNING | omega.estimator.assemble - No mutations found for ['P19_0044_BDO_01'], ['essential_splice'], ['STAG2'], filling the counts with 0s.
2024-04-10 17:55:47,693 - WARNING | omega.estimator.assemble - No mutations found for ['P19_0044_BDO_01'], ['nonsense'], ['TERT'], filling the counts with 0s.
2024-04-10 17:55:47,694 - WARNING | omega.estimator.assemble - No mutations found for ['P19_0044_BDO_01'], ['essential_splice'], ['TERT'], filling the counts with 0s.
2024-04-10 17:55:47,694 - WARNING | omega.estimator.assemble - Lambda tensors are 0, we are ignoring this case ['P19_0044_BDO_01'], ['essential_splice'], ['TERT'].
2024-04-10 17:55:47,696 - WARNING | omega.estimator.assemble - No mutations found for ['P19_0044_BDO_01'], ['nonsense'], ['TP53'], filling the counts with 0s.
2024-04-10 17:55:47,698 - WARNING | omega.estimator.assemble - No mutations found for ['P19_0044_BDO_01'], ['essential_splice'], ['TP53'], filling the counts with 0s.
2024-04-10 17:55:47,730 - INFO    | omega.estimator - Data prepared

```

AFTER:
```
2024-04-10 18:19:51,290 - WARNING | omega.estimator.assemble - No mutations found for ['P19_0044_BDO_01'], ['nonsense'], ['ARID1A'], filling the counts with 0s.
2024-04-10 18:19:51,291 - WARNING | omega.estimator.assemble - No mutations found for ['P19_0044_BDO_01'], ['essential_splice'], ['ARID1A'], filling the counts with 0s.
2024-04-10 18:19:51,294 - WARNING | omega.estimator.assemble - No mutations found for ['P19_0044_BDO_01'], ['nonsense'], ['CDKN1A'], filling the counts with 0s.
2024-04-10 18:19:51,295 - WARNING | omega.estimator.assemble - No mutations found for ['P19_0044_BDO_01'], ['essential_splice'], ['CDKN1A'], filling the counts with 0s.
2024-04-10 18:19:51,297 - WARNING | omega.estimator.assemble - No mutations found for ['P19_0044_BDO_01'], ['nonsense'], ['CREBBP'], filling the counts with 0s.
2024-04-10 18:19:51,298 - WARNING | omega.estimator.assemble - No mutations found for ['P19_0044_BDO_01'], ['essential_splice'], ['CREBBP'], filling the counts with 0s.
2024-04-10 18:19:51,300 - WARNING | omega.estimator.assemble - No mutations found for ['P19_0044_BDO_01'], ['essential_splice'], ['EP300'], filling the counts with 0s.
2024-04-10 18:19:51,302 - WARNING | omega.estimator.assemble - No mutations found for ['P19_0044_BDO_01'], ['missense'], ['FGFR3'], filling the counts with 0s.
2024-04-10 18:19:51,303 - WARNING | omega.estimator.assemble - Lambda tensors are 0, we are ignoring this case ['P19_0044_BDO_01'], ['missense'], ['FGFR3'].
2024-04-10 18:19:51,303 - WARNING | omega.estimator.assemble - No mutations found for ['P19_0044_BDO_01'], ['nonsense'], ['FGFR3'], filling the counts with 0s.
2024-04-10 18:19:51,304 - WARNING | omega.estimator.assemble - Lambda tensors are 0, we are ignoring this case ['P19_0044_BDO_01'], ['nonsense'], ['FGFR3'].
2024-04-10 18:19:51,304 - WARNING | omega.estimator.assemble - No mutations found for ['P19_0044_BDO_01'], ['essential_splice'], ['FGFR3'], filling the counts with 0s.
2024-04-10 18:19:51,305 - WARNING | omega.estimator.assemble - Lambda tensors are 0, we are ignoring this case ['P19_0044_BDO_01'], ['essential_splice'], ['FGFR3'].
2024-04-10 18:19:51,305 - WARNING | omega.estimator.assemble - No mutations found for ['P19_0044_BDO_01'], ['missense', 'nonsense'], ['FGFR3'], filling the counts with 0s.
2024-04-10 18:19:51,306 - WARNING | omega.estimator.assemble - Lambda tensors are 0, we are ignoring this case ['P19_0044_BDO_01'], ['missense', 'nonsense'], ['FGFR3'].
2024-04-10 18:19:51,306 - WARNING | omega.estimator.assemble - No mutations found for ['P19_0044_BDO_01'], ['missense', 'nonsense', 'essential_splice'], ['FGFR3'], filling the counts with 0s.
2024-04-10 18:19:51,307 - WARNING | omega.estimator.assemble - Lambda tensors are 0, we are ignoring this case ['P19_0044_BDO_01'], ['missense', 'nonsense', 'essential_splice'], ['FGFR3'].
2024-04-10 18:19:51,308 - WARNING | omega.estimator.assemble - No mutations found for ['P19_0044_BDO_01'], ['nonsense'], ['FOXQ1'], filling the counts with 0s.
2024-04-10 18:19:51,309 - WARNING | omega.estimator.assemble - No mutations found for ['P19_0044_BDO_01'], ['essential_splice'], ['FOXQ1'], filling the counts with 0s.
2024-04-10 18:19:51,309 - WARNING | omega.estimator.assemble - Lambda tensors are 0, we are ignoring this case ['P19_0044_BDO_01'], ['essential_splice'], ['FOXQ1'].
2024-04-10 18:19:51,311 - WARNING | omega.estimator.assemble - No mutations found for ['P19_0044_BDO_01'], ['essential_splice'], ['KDM6A'], filling the counts with 0s.
2024-04-10 18:19:51,313 - WARNING | omega.estimator.assemble - No mutations found for ['P19_0044_BDO_01'], ['nonsense'], ['KMT2C'], filling the counts with 0s.
2024-04-10 18:19:51,318 - WARNING | omega.estimator.assemble - No mutations found for ['P19_0044_BDO_01'], ['essential_splice'], ['NOTCH2'], filling the counts with 0s.
2024-04-10 18:19:51,320 - WARNING | omega.estimator.assemble - Lambda tensors are 0, we are ignoring this case ['P19_0044_BDO_01'], ['missense'], ['PIK3CA'].
2024-04-10 18:19:51,320 - WARNING | omega.estimator.assemble - No mutations found for ['P19_0044_BDO_01'], ['nonsense'], ['PIK3CA'], filling the counts with 0s.
2024-04-10 18:19:51,321 - WARNING | omega.estimator.assemble - Lambda tensors are 0, we are ignoring this case ['P19_0044_BDO_01'], ['nonsense'], ['PIK3CA'].
2024-04-10 18:19:51,321 - WARNING | omega.estimator.assemble - No mutations found for ['P19_0044_BDO_01'], ['essential_splice'], ['PIK3CA'], filling the counts with 0s.
2024-04-10 18:19:51,322 - WARNING | omega.estimator.assemble - Lambda tensors are 0, we are ignoring this case ['P19_0044_BDO_01'], ['essential_splice'], ['PIK3CA'].
2024-04-10 18:19:51,323 - WARNING | omega.estimator.assemble - Lambda tensors are 0, we are ignoring this case ['P19_0044_BDO_01'], ['missense', 'nonsense'], ['PIK3CA'].
2024-04-10 18:19:51,324 - WARNING | omega.estimator.assemble - Lambda tensors are 0, we are ignoring this case ['P19_0044_BDO_01'], ['missense', 'nonsense', 'essential_splice'], ['PIK3CA'].
2024-04-10 18:19:51,324 - WARNING | omega.estimator.assemble - No mutations found for ['P19_0044_BDO_01'], ['missense'], ['RB1'], filling the counts with 0s.
2024-04-10 18:19:51,325 - WARNING | omega.estimator.assemble - No mutations found for ['P19_0044_BDO_01'], ['nonsense'], ['RB1'], filling the counts with 0s.
2024-04-10 18:19:51,325 - WARNING | omega.estimator.assemble - No mutations found for ['P19_0044_BDO_01'], ['essential_splice'], ['RB1'], filling the counts with 0s.
2024-04-10 18:19:51,326 - WARNING | omega.estimator.assemble - No mutations found for ['P19_0044_BDO_01'], ['missense', 'nonsense'], ['RB1'], filling the counts with 0s.
2024-04-10 18:19:51,327 - WARNING | omega.estimator.assemble - No mutations found for ['P19_0044_BDO_01'], ['missense', 'nonsense', 'essential_splice'], ['RB1'], filling the counts with 0s.
2024-04-10 18:19:51,328 - WARNING | omega.estimator.assemble - No mutations found for ['P19_0044_BDO_01'], ['nonsense'], ['RBM10'], filling the counts with 0s.
2024-04-10 18:19:51,329 - WARNING | omega.estimator.assemble - No mutations found for ['P19_0044_BDO_01'], ['essential_splice'], ['RBM10'], filling the counts with 0s.
2024-04-10 18:19:51,330 - WARNING | omega.estimator.assemble - Lambda tensors are 0, we are ignoring this case ['P19_0044_BDO_01'], ['missense'], ['STAG2'].
2024-04-10 18:19:51,331 - WARNING | omega.estimator.assemble - Lambda tensors are 0, we are ignoring this case ['P19_0044_BDO_01'], ['nonsense'], ['STAG2'].
2024-04-10 18:19:51,332 - WARNING | omega.estimator.assemble - No mutations found for ['P19_0044_BDO_01'], ['essential_splice'], ['STAG2'], filling the counts with 0s.
2024-04-10 18:19:51,332 - WARNING | omega.estimator.assemble - Lambda tensors are 0, we are ignoring this case ['P19_0044_BDO_01'], ['essential_splice'], ['STAG2'].
2024-04-10 18:19:51,333 - WARNING | omega.estimator.assemble - Lambda tensors are 0, we are ignoring this case ['P19_0044_BDO_01'], ['missense', 'nonsense'], ['STAG2'].
2024-04-10 18:19:51,334 - WARNING | omega.estimator.assemble - Lambda tensors are 0, we are ignoring this case ['P19_0044_BDO_01'], ['missense', 'nonsense', 'essential_splice'], ['STAG2'].
2024-04-10 18:19:51,335 - WARNING | omega.estimator.assemble - Lambda tensors are 0, we are ignoring this case ['P19_0044_BDO_01'], ['missense'], ['TERT'].
2024-04-10 18:19:51,335 - WARNING | omega.estimator.assemble - No mutations found for ['P19_0044_BDO_01'], ['nonsense'], ['TERT'], filling the counts with 0s.
2024-04-10 18:19:51,336 - WARNING | omega.estimator.assemble - Lambda tensors are 0, we are ignoring this case ['P19_0044_BDO_01'], ['nonsense'], ['TERT'].
2024-04-10 18:19:51,336 - WARNING | omega.estimator.assemble - No mutations found for ['P19_0044_BDO_01'], ['essential_splice'], ['TERT'], filling the counts with 0s.
2024-04-10 18:19:51,337 - WARNING | omega.estimator.assemble - Lambda tensors are 0, we are ignoring this case ['P19_0044_BDO_01'], ['essential_splice'], ['TERT'].
2024-04-10 18:19:51,338 - WARNING | omega.estimator.assemble - Lambda tensors are 0, we are ignoring this case ['P19_0044_BDO_01'], ['missense', 'nonsense'], ['TERT'].
2024-04-10 18:19:51,339 - WARNING | omega.estimator.assemble - Lambda tensors are 0, we are ignoring this case ['P19_0044_BDO_01'], ['missense', 'nonsense', 'essential_splice'], ['TERT'].
2024-04-10 18:19:51,339 - WARNING | omega.estimator.assemble - Lambda tensors are 0, we are ignoring this case ['P19_0044_BDO_01'], ['missense'], ['TP53'].
2024-04-10 18:19:51,340 - WARNING | omega.estimator.assemble - No mutations found for ['P19_0044_BDO_01'], ['nonsense'], ['TP53'], filling the counts with 0s.
2024-04-10 18:19:51,340 - WARNING | omega.estimator.assemble - Lambda tensors are 0, we are ignoring this case ['P19_0044_BDO_01'], ['nonsense'], ['TP53'].
2024-04-10 18:19:51,341 - WARNING | omega.estimator.assemble - No mutations found for ['P19_0044_BDO_01'], ['essential_splice'], ['TP53'], filling the counts with 0s.
2024-04-10 18:19:51,341 - WARNING | omega.estimator.assemble - Lambda tensors are 0, we are ignoring this case ['P19_0044_BDO_01'], ['essential_splice'], ['TP53'].
2024-04-10 18:19:51,342 - WARNING | omega.estimator.assemble - Lambda tensors are 0, we are ignoring this case ['P19_0044_BDO_01'], ['missense', 'nonsense'], ['TP53'].
2024-04-10 18:19:51,343 - WARNING | omega.estimator.assemble - Lambda tensors are 0, we are ignoring this case ['P19_0044_BDO_01'], ['missense', 'nonsense', 'essential_splice'], ['TP53'].
2024-04-10 18:19:51,362 - INFO    | omega.estimator - Data prepared

```

**Comment by FerriolCalvet** [comment]
@koszulordie and @FerriolCalvet agreed that this is an appropriate fix for these cases where the mutabilities are 0.

Closed with commit 2e7561d30c28597e6f56432f2f707de0e0e4b084
- [x] works with MLE
- [x] works with Bayes
- [x] single sample
- [x] all samples
